File: lib/domain/resource.dart
enum ResourceType {
  YOUTUBE,
  PDF,
  IMAGE,
  VIDEO,
  FILE,
  LINK,
  UNKNOWN;

  static ResourceType fromString(String type) {
    return ResourceType.values.firstWhere(
      (e) => e.name == type.toUpperCase(),
      orElse: () => ResourceType.UNKNOWN,
    );
  }
}

class Resource {
  final int id;
  final String title;
  final String resourceUrl;
  final ResourceType type;
  final String? description;
  final DateTime createdAt;
  final int noteId;

  Resource({
    required this.id,
    required this.title,
    required this.resourceUrl,
    required this.type,
    this.description,
    required this.createdAt,
    required this.noteId,
  });

  factory Resource.fromJson(Map<String, dynamic> json) {
    return Resource(
      id: json['id'],
      title: json['title'],
      resourceUrl: json['resourceUrl'],
      type: ResourceType.fromString(json['type']),
      description: json['description'],
      createdAt: DateTime.parse(json['createdAt']),
      noteId: json['noteId'],
    );
  }
}

File: eduvisionbackend/src/main/java/com/academicproject/eduvisionbackend/entity/Resource.java
package com.academicproject.eduvisionbackend.entity;

import java.time.LocalDateTime;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "resources")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Resource {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    // YouTube URL / File Path / CDN URL
    @Column(nullable = false, columnDefinition = "TEXT")
    private String resourceUrl;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ResourceType type;

    private String description;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "note_id", nullable = false)
    private Note note;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum ResourceType {
        YOUTUBE,
        PDF,
        IMAGE,
        VIDEO,
        FILE,
        LINK
    }
}